cmd/clusterawsadm/ami/helper.go

```diff
@@ -58,27 +61,54 @@ func getimageRegionList() []string {
 	}
 }
 
-func getSupportedKubernetesVersions() ([]string, error) {
+// getSupportedKubernetesVersions returns all possible k8s versions till last nth kubernetes release.
+func getSupportedKubernetesVersions(lastNReleases int) ([]string, error) {
 	supportedVersions := make([]string, 0)
-	latestVersion, err := latestStableRelease()
+	currentVersion, err := latestStableRelease()
 	if err != nil {
 		return nil, err
 	}
 
-	supportedVersions = append(supportedVersions, latestVersion)
-	nMinusOne, err := kubernetesversions.PreviousMinorRelease(latestVersion)
+	patchVersions, err := allPatchVersions(currentVersion)
 	if err != nil {
 		return nil, err
 	}
-	supportedVersions = append(supportedVersions, nMinusOne)
 
-	nMinusTwo, err := kubernetesversions.PreviousMinorRelease(nMinusOne)
+	supportedVersions = append(supportedVersions, currentVersion)
+	supportedVersions = append(supportedVersions, patchVersions...)
+
```

Comment:
```suggestion
	patchVersions = append(supportedVersions, currentVersion)
```

Reply:
I am returning supportedVersions from func, and first I will currentVersion and then add related patch versions in supportedVersions

> patchVersions = append(supportedVersions, currentVersion)

I am not able to understand the purpose of this suggestion, with this way, will loose all patch versions and supportedVersions is still empty

Reply:
Sorry I meant 
`patchVersions = append(patchVersions, currentVersion)` to avoid having a new `supportedVersions` variable.

Reply:
done